### ISSUE TITLE
ONNX inference is too slow

### ISSUE THREAD

**Issue by drcodex** [Deci-AI/super-gradients#1098]
### 💡 Your Question

I trained a custom dataset on top of pre-trained weights, ONNX inference time seems too much compared to torch model.
Torch : 0.2 seconds
onnx: 4.5 seconds
I understand that onnx could be slower, but this difference seems too much. Any thoughts?

**Conversion script that I used:** 

from super_gradients.training import models
from super_gradients.training.utils.quantization.export import export_quantized_module_to_onnx
from super_gradients.training.utils.quantization.selective_quantization_utils import SelectiveQuantizer

MODEL_ARCH = 'yolo_nas_m'
model_path="ckpt_best.pth"
model = models.get(MODEL_ARCH, checkpoint_path=model_path,num_classes=12)

q_util = SelectiveQuantizer(
    default_quant_modules_calibrator_weights="max",
    default_quant_modules_calibrator_inputs="histogram",
    default_per_channel_quant_weights=True,
    default_learn_amax=False,
    verbose=True,
)
q_util.quantize_module(model)
model.prep_model_for_conversion(input_size=[1, 3, 640, 640])

# Convert model to onnx
onnx_filename = "out.onnx"

export_quantized_module_to_onnx(
    model=model.cpu(),
    onnx_filename=onnx_filename,
    input_shape=[1, 3, 640, 640],
    input_size=[1, 3, 640, 640],
    verbose=True,
)

**Inference code using onnx:** 

import cv2

img = cv2.imread(image)
img = cv2.resize(img, dsize=(640, 640), interpolation=cv2.INTER_AREA)
img.resize((1, 3, 640, 640),refcheck=False)
data = json.dumps({'data': img.tolist()})
data = np.array(json.loads(data)['data']).astype('float32')

onnx_inputs= {onnx_session.get_inputs()[0].name:data}
onnx_output = onnx_session.run(None, onnx_inputs)

**Comment by NatanBagrov** [comment]
Hi, @drcodex and thanks for using SG!
This is expected due to overhead of Q/DQ layers. Please read [this tutorial](https://github.com/Deci-AI/super-gradients/blob/master/documentation/source/BenchmarkingYoloNAS.md) carefully.

**Comment by drcodex** [comment]
@NatanBagrov Are you saying that inference with onnxruntime will always be slow? and the recommended deployment framework is tensorRT, for optimum inference speed?

**Comment by mikel-brostrom** [comment]
You won't see INT8 model boosts compared to Float32 on Intel CPUs under 10th gen. This is because Intel CPUs < 10th gen don't have Intel DLBoost, a specific instruction set (ISA) architecture designed to improve performance of INT8 DL models. This ISA is present in Intel chips from 10th gen onwards. It is my understanding that, without a specific INT8 ISA, the operations get upsampled to Float32.

**Comment by NatanBagrov** [comment]
@drcodex, slow is a relative thing.
I believe on FP16 it should be faster.
Your ONNX has Q/DQ layers, and it is FP32, the quantization itself (to INT8) happens on the specific HW during the compilation to/with TensorRT
